File: cpp/172FactorialTrailingZeroes/172FactorialTrailingZeroes.cpp
#include <sstream>
#include <iostream>
#include <string>
#include <vector>
#include <stack>
#include <unordered_map>
#include <algorithm>
#include <limits>

using namespace std;

/*
Given an integer n, return the number of trailing zeroes in n!.

Note: Your solution should be in logarithmic time complexity.

Credits:
Special thanks to @ts for adding this problem and creating all test cases.
*/

class Solution {
public:
    int trailingZeroes(int n) {
        int sum = 0;
        do{
        	n /= 5;
        	sum+=n;
        }while(n);

        return sum;
    }
};

int main(){
	Solution s;

	return 0;
}

File: src/172FactorialTrailingZeroes/172FactorialTrailingZeroes.go
package main

import "fmt"

func trailingZeroes(n int) int {
	sum := 0
	for n != 0 {
		sum += n / 5
		n /= 5
	}
	return sum
}

func main() {
	fmt.Printf("%v\n", trailingZeroes(1808548329))
	fmt.Printf("%v\n", trailingZeroes(100))
}
